ops/pebble.py

```diff
@@ -1225,3 +1508,225 @@ def remove_path(self, path: str, *, recursive: bool = False):
+
+        Most of the parameters are explained in the "Parameters" section
+        below, however, input/output handling is a bit more complex. Some
+        examples are shown below::
+
+            # Simple command with no output; just check exit code
+            >>> process = client.exec(['send-emails'])
+            >>> process.wait()
+
+            # Fetch output as string
+            >>> process = client.exec(['python3', '--version'])
+            >>> version, _ = process.wait_output()
+            >>> print(version)
+            Python 3.8.10
+
+            # Fetch both stdout and stderr as strings
+            >>> process = client.exec(['pg_dump', '-s', ...])
+            >>> schema, logs = process.wait_output()
+
+            # Stream input from a string and write output to files
+            >>> in = 'foo\nbar\n'
+            >>> with open('out.txt', 'w') as out, open('err.txt', 'w') as err:
+            ...     process = client.exec(['awk', '{ print toupper($0) }'],
+            ...                           stdin=in, stdout=out, stderr=err)
+            ...     process.wait()
+            >>> open('out.txt').read()
+            'FOO\nBAR\n'
+            >>> open('err.txt').read()
+            ''
+
+            # Real-time streaming using ExecProcess.stdin and ExecProcess.stdout
+            >>> process = client.exec(['cat'])
+            >>> def stdin_thread():
+            ...     for line in ['one\n', '2\n', 'THREE\n']:
+            ...         process.stdin.write(line)
+            ...         process.stdin.flush()
+            ...         time.sleep(1)
+            ...     process.stdin.close()
+            ...
+            >>> threading.Thread(target=stdin_thread).start()
+            >>> for line in process.stdout:
+            ...     print(datetime.datetime.now().strftime('%H:%M:%S'), repr(line))
+            ...
+            16:20:26 'one\n'
+            16:20:27 '2\n'
+            16:20:28 'THREE\n'
+            >>> process.wait()  # will return immediately as stdin was closed
+
+            # Show exception raised for non-zero return code
+            >>> process = client.exec(['ls', 'notexist'])
+            >>> out, err = process.wait_output()
+            Traceback (most recent call last):
+              ...
+            ExecError: "ls" returned exit code 2
+            >>> exc = sys.last_value
+            >>> exc.exit_code
+            2
+            >>> exc.stdout
+            ''
+            >>> exc.stderr
+            "ls: cannot access 'notfound': No such file or directory\n"
+
+        Args:
+            command: Command to execute: the first item is the name (or path)
+                of the executable, the rest of the items are the arguments.
+            environment: Environment variables to pass to the process.
+            working_dir: Working directory to run the command in. If not set,
+                Pebble uses the target user's $HOME directory.
```

Comment:
and if no user is specified? does it use the pebble cwd?

Reply:
Thanks - clarified in the doc comment.